Tests/SentryTests/Integrations/SessionReplay/SentryOnDemandReplayTests.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import Nimble
+@testable import Sentry
+import SentryTestUtils
+import XCTest
+
+#if os(iOS) || os(tvOS)
+class SentryOnDemandReplayTests: XCTestCase {
+    
+    private let workingQueue = DispatchQueue(label: "io.sentry.TestQueue")
+    
+    func getSut() -> SentryOnDemandReplay {
+        let outputPath = FileManager.default.temporaryDirectory
+        let sut = SentryOnDemandReplay(outputPath: outputPath.path)
+        sut.workingQueue = workingQueue
+        return sut
+    }
+    
+    func testAddFrame() {
+        let sut = getSut()
+        sut.addFrameAsync(image: UIImage.add)
+       
+        waitWorkingQueue()
+        
+        guard let frame = sut.frames.first else {
+            fail("Frame was not saved")
+            return
+        }
+        expect(FileManager.default.fileExists(atPath: frame.imagePath)) == true
```

Comment:
`m`: If I got it right, other classes rely on the sut writing files to the imagePath you pass in when initializing the sut. It would be great if this test validated the behavior of the class in the same way as other classes use it. Meaning instead of exposing the frames to get the image path, we could check if a file exists at the path that we pass to the constructor of the sut or we can test the behavior similarly as we already do in testGenerateVideo.

Reply:
The output instruction helps the class to do a clean up latter. But the classes that use sut dont read the files directly. 

But I agree that we should check whether the sut is writing in the right place.